source/material_model/rheology/visco_plastic.cc

```diff
@@ -403,17 +404,15 @@ namespace aspect
                    ExcMessage("Invalid strain_rate in the MaterialModelInputs. This is likely because it was "
                               "not filled by the caller."));
 
+            const SymmetricTensor<2,dim> deviatoric_strain_rate = deviator(in.strain_rate[i]);
+
             // For each independent component, compute the derivative.
             for (unsigned int component = 0; component < SymmetricTensor<2,dim>::n_independent_components; ++component)
               {
                 const TableIndices<2> strain_rate_indices = SymmetricTensor<2,dim>::unrolled_to_component_indices (component);
 
-                // components that are not on the diagonal are multiplied by 0.5, because the symmetric tensor
-                // is modified by 0.5 in both symmetric directions (xy/yx) simultaneously and we compute the combined
-                // derivative
-                const SymmetricTensor<2,dim> strain_rate_difference = in.strain_rate[i]
-                                                                      + std::max(std::fabs(in.strain_rate[i][strain_rate_indices]), min_strain_rate)
-                                                                      * (component > dim-1 ? 0.5 : 1 )
```

Comment:
I think the $\times$ 0.5 here is redundant, because the $\times$ 0.5 has already been applied internally in function  symmetrize() called by Utilities::nth_basis_for_symmetric_tensors()

Reply:
Great!